monitoring/prometheus-rules/vm.yaml

```
@@ -0,0 +1,21 @@
+- name: kubernetes.virtualization.vm
+  rules:
+    - alert: D8VirtualizationVMFirmwareOutOfDate
+      expr: d8_virtualization_virtualmachine_firmware_up_to_date == 0
+      labels:
+        severity_level: "6"
+        tier: cluster
+      for: 30m
+      annotations:
+        plk_protocol_version: "1"
+        plk_markup_format: "markdown"
+        plk_create_group_if_not_exists__d8_virtualization_vm_firmware_out_of_date: "D8VirtualizationVMFirmwareOutOfDate,tier=~tier,prometheus=deckhouse,kubernetes=~kubernetes"
+        plk_grouped_by__d8_virtualization_vm_firmware_out_of_date: "D8VirtualizationVMFirmwareOutOfDate,tier=~tier,prometheus=deckhouse,kubernetes=~kubernetes"
+        summary: VirtualMachine firmware is out of date.
+        description: |
+          The recommended course of action:
+          Find VirtualMachines whose firmware is out of date:
+          ```
+          kubectl get vm -A -o jsonpath="{range .items[?(@.status.conditions)]}{.metadata.namespace}{'\t'}{.metadata.name}{'\t'}{range .status.conditions[?(@.type=='FirmwareUpToDate')]}{.status}{'\n'}{end}{end}" | awk '$3 != "True" {print $1 "\t" $2}'
+          ```
+          Then update the firmware according to the documentation or contact the support team.
```

```suggestion
          To find VirtualMachines that have outdated firmware run the following command:
          ```
          kubectl get vm -A -o jsonpath="{range .items[?(@.status.conditions)]}{.metadata.namespace}{'\t'}{.metadata.name}{'\t'}{range .status.conditions[?(@.type=='FirmwareUpToDate')]}{.status}{'\n'}{end}{end}" | awk '$3 != "True" {print $1 "\t" $2}'
          ```
          The VirtualMachine firmware is updated automatically after a new module is installed. To perform the procedure manually, evict the VM to a new node, or reboot it.
```